mwclient/client.py

```diff
@@ -892,6 +892,8 @@ def upload(self, file=None, filename=None, description='', ignore=False,
                            stashed temporarily.
             comment (str): Upload comment. Also used as the initial page text
                            for new files if `description` is not specified.
+            asynchronous (str): Whether the server should make large file
```

Comment:
The type of the `asynchronous` argument should be `bool` instead of `str`.
```suggestion
            asynchronous (bool): Whether the server should make large file
```